space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+
+### Space Manager 
+The Space Manager (SM) is a Kuberentes controller that is responsible for maintaing the state of the Space objects. The SM reconcile the Space and SpaceProviderDesc objects, and monitor the state of the pSpaces through the space provider.   
+The SM uses a library of space provider adaptors to communicate with the space providers. Currently the SF includes 3 space provider adaptors KCP, KIND, and KubeFlex.
+
+### Space provider adaptors
+The space manager uses a set of space provider adaptors that interacts with the space provider. All space provider adaptors implement a simple "space client interface" that includes basic pSpace life cycle operatios. In addition the interface also defines the events that sent from the adaptors to the space manager.
+
+### Space aware client (SAC)
+The Space aware client (msclient) allows clients/controllers to easily get access to the underlying space by simply using the space name. There is no direct interaction between the msclient and the Space  provider, and therefore the msclient is transparent to the specific provider of the spaces. 
+![Space Aware Client](SAC.drawio.svg "Space Aware Client")
+The only requirement is that the space can be accessed through regular Kube APIs when using the appropriate kubeconfig information. 
+
+**Main features**  
+- Constantly watch for changes in the available spaces. The SAC is using an informer on Space objecs using the SF Space client-go APIs.
+    - Holds a cache of the pSpace access info for each Space 
+- Exposes utility functions to get RestConfig by simpy specifying the space name  
+
+Currently the SAC exposes a simple function that supply the config access info for a space. In the future we will add a SAC space aware ClientSets
+
+## Architecture details 
+When a new SpaceProviderDesc is created the SM creates a new namespace that will be used to host all the Space objects representing pSpaces on that space provider.  
+The name of the generated namespace will be `paceprovider-<provider name>`
+
+![Space, SpaceProviderDesc and the Space provider namespace](SF-NS.drawio.svg "SF Namespaces")
+
+Note: In the future the spaces will be created in their own namespace and not in namespace per provider. 
+
+The Space also holds a reference to the SpaceProviderDesc associated with this Space and some capabilities (e.g., the space type). The status of the space holds a secret reference to the secret holding the access information for that pSpace. As described before for managed spaces the SF retrive that information from teh space provider, create the needed secrets and update the Space object with the secret reference
+
```

Comment:
Somewhere around here there needs to be a discussion of the mapping between the name of the Space object and the relevant provider's identifier(s) for the corresponding physical space. For example, in the kcp provider there are two identifiers: the `logicalcluster.Name` that lies in a flat namespace and there is the pathname of the corresponding workspace (when there is one). It seems to me that this mapping (really, these mappings --- one for each sort of identifier the provider has) is a degree of freedom that the provider adapter has, and thus needs to be documented for each provider adapter.

Reply:
Correct. We still not have that mapping well defined in the SM and we use Space name == pSpace name . We definetly need to maintain the mapping between the Space name and the  provider's pSpace ID

Reply:
I find saying "Space name == pSpace name" to be confusing.  I think of the physical space as the actual thing that the provider provides. So its name or names or IDs are the names/IDs that the provider uses. Which I think you agreed is not necessarily the same as the name of the Space object.

Reply:
Currently the "Space Name" == "the pSpace identifier used by the space provider"

Reply:
Ah, perhaps you are using "the pSpace identifier used by the space provider" to mean some particular technical thing that I am not aware of.  It is an identifier in the interface between Space Manager and provider adapter?

Reply:
No, it is identifyer between the adaptor and the space provider itself (or the iterface it exposes) For example, for KIND provider the KIND cluster name is the id of the pSpace. and as stated above currently the space name will be used as the KIND cluster name

Reply:
In the kcp case, what is "the pSpace identifier used by the space provider"? kcp identifies its spaces in two ways: `logicalcluster.Name` and workspace pathname. So I find an assertion that implies there is only to be confounding.

Reply:
for KCP we use th eWorkspace path name

Reply:
A kcp workspace pathname is a structured thing in the abstract, with a string representation using colon (`:`) to separate parts of the path. Are you saying that when using a kcp provider, the Space object name will be a kcp workspace pathname, such as "root:espw"?

Reply:
I wasn't accurate. The space name isused as the workspace object's name  (in meta) .

Reply:
You finally stated the missing information for the kcp provider adapter. It is a choice made by that provider adapter, about the relation it maintains between Space object name and the ways that kcp identifies its logicalclusters. My point is that in general this is a question that has to be answered (i.e., documented) for each provider adapter: the relation it maintains between Space object name and the ways that the provider identifies its native cluster-ish thingies.

Reply:
Correct. I'll add a section on "Provider Adaptors"  And will put there the details for a specific provider adaptor